api/service/types.ts

```diff
@@ -46,59 +50,61 @@ export interface CarStoreBucket {
 }
 
 export interface StoreTable {
-  exists: (uploaderDID: string, payloadCID: string) => Promise<boolean>
+  exists: (space: DID, link: AnyLink) => Promise<boolean>
   insert: (item: StoreItemInput) => Promise<StoreItemOutput>
-  remove: (uploaderDID: string, payloadCID: string) => Promise<void>
-  list: (uploaderDID: string, options?: ListOptions) => Promise<ListResponse<StoreListResult>>
+  remove: (space: DID, link: AnyLink) => Promise<void>
+  list: (space: DID, options?: ListOptions) => Promise<ListResponse<StoreListResult>>
 }
 
 export interface UploadTable {
-  exists: (uploaderDID: string, dataCID: string) => Promise<boolean>
-  insert: (uploaderDID: string, item: UploadItemInput) => Promise<UploadItemOutput[]>
-  remove: (uploaderDID: string, dataCID: string) => Promise<void>
-  list: (uploaderDID: string, options?: ListOptions) => Promise<ListResponse<UploadItemOutput>>
+  exists: (space: DID, root: AnyLink) => Promise<boolean>
+  insert: (item: UploadItemInput) => Promise<UploadItemOutput[]>
+  remove: (space: DID, root: AnyLink) => Promise<void>
+  list: (space: DID, options?: ListOptions) => Promise<ListResponse<UploadItemOutput>>
 }
```

Comment:
So, my previous approach was having tables/* functions to have the types in the schema. Being caller responsibility to send strings instead of link/did/...

I think we should go with previous way of doing things to have DB layer clean and not responsible to know what formats UCAN have. Even more as we did not go with UCAN flavour schema

Finally, it would keep input and outputs the same in database layer, which with current commit does not happen

Reply:
note: We've ended up with a schema that is basically the ucan flavour! 

I was thinking that this table wrapper could take in the data as we have it, as DIDs and CIDs, and do the work of preparing it for the database. That we we get stronger type assertions on the method signatures, instead of `string` all the things` which would make it easier to mix up the args and not get a warning. There is more to do there as we still loose type info on things like the responses from dynamo, but I stand by this general pattern of having the wrapper map "the data as we have it" to the "the data as the db needs it".

I recall @alanshaw asked for this change on a previous PR as well!

Reply:
> note: We've ended up with a schema that is basically the ucan flavour!

In a way, I think we ended up in a between domain and UCAN based on initial proposal. 

I like to keep things separated and also consider it to be the best future proof solution. If we move to a world where CF Hosted API is `upload-api` implementing UCAN server, together with a Gateway with a Dynamo behind, we will get things written to Dynamo from a HTTP gateway that would not require data marshalling to move DID/CID around, instead of strings like the Data Model. If we move to a future where DB lives in different microservice, same applies where it should only know about its data model.

I don't remember @alanshaw calling this out before, or I did not understand that. Let's get Alan's input to get a decision

Reply:
I was referring to something slightly different - the response data from UCAN invocations can have CIDs because it's serialized using IPLD. It's just one less thing to serialize/deserialize (*serde*) for the wire.

IMO this abstraction should do the job of serde and we should work with rich types as much as possible. If this abstraction takes strings then we push serde code into multiple places in the code base or have a utility function to do the job and use that in multiple places - why not just have the abstraction do that? When do we want to use that utility for a purpose other than to prepare input for the abstraction?

Whenever we have to do serde for wire transfer we have to validate the serialized data the other end - we can't just pump it direct into dynamo. It usually involves parsing to create the rich data structure regardless.

Reply:
one route handler may call multiple of this abstractions. between tables and bucket abstractions, moving this direction means all abstractions need to handle transforming things themselves rather than the caller. 

I am ok moving forward this way, just raising my concerns with the decision